warehouse/core/src/test/java/datawave/iterators/filter/ageoff/FieldAgeOffFilterTest.java

```diff
@@ -423,11 +425,11 @@ public void handlesFieldsStartingWithNumber() {
         ageOffFilter.init(filterOptions, iterEnv);
         // field_y is a match, but its ttl was not defined, so it will use the default one
         Key key = new Key("1234", "myDataType\\x00my-uuid", "field_y\u0000value", VISIBILITY_PATTERN, tenSecondsAgo);
-        Assert.assertFalse(ageOffFilter.accept(filterOptions.getAgeOffPeriod(System.currentTimeMillis()), key, new Value()));
+        Assert.assertFalse(ageOffFilter.accept(filterOptions.getAgeOffPeriod(currentTime), key, new Value()));
         Assert.assertTrue(ageOffFilter.isFilterRuleApplied());
         // field_z is a match, but the key is more recent than its age off period
         Key keyNumeric = new Key("1234", "myDataType\\x00my-uuid", "12_3_4\u0000value", VISIBILITY_PATTERN, oneSecondAgo);
-        Assert.assertTrue(ageOffFilter.accept(filterOptions.getAgeOffPeriod(System.currentTimeMillis()), keyNumeric, new Value()));
+        Assert.assertTrue(ageOffFilter.accept(filterOptions.getAgeOffPeriod(currentTime), keyNumeric, new Value()));
```

Comment:
This assertion seems to fail sporadically due to the interplay between `oneSecondAgo`, the 5sec TTL, and the amount of time that it takes for this test to run, which is subject to the whims of the test runner JVM. E.g., you can force the failure simply by placing a breakpoint at the top of this test and slowly stepping through.

Moving the `FieldAgeOffFilter` creation, which seems to be fairly expensive, to the top and pinning assertions to the `currentTime` definition seems to help, in my testing at least

Reply:
+1